features/0434-outofband/README.md

```diff
@@ -0,0 +1,417 @@
+- A website operator wants to request something from a user on their site where there is no (known) connection with the user.
+
+In both cases, there is only a single out of band protocol message sent. The message responding to the out of band message is a DIDComm message from an appropriate protocol.
+
+Note that the website-to-agent model is not the only such interaction enabled by the out of band protocol, and a QR code is not the only delivery mechanism for out of band messages. They are useful as examples of the purpose of the protocol.
+
+### Roles
+
+The out of band protocol has two roles: __sender__ and __receiver__.
+
+#### sender
+
+The agent that generates the out of band message and makes it available to the other party.
+
+#### receiver
+
+The agent that receives the out of band message and decides how to respond. There is no out of band protocol message with which the receiver will respond. Rather, if they respond, they will use a message from another protocol that the sender will understand.
+
+### States
+
+The state machines for the sender and receiver are a bit odd for the out of band protocol because of it being a single message protocol with a response from a different protocol. In the following state machine diagrams we generically describe the response message from the *receiver* as being a DIDComm message without associating the message with a specific protocol.
+
+The *sender* state machine is as follows:
+
+[![sender state machine google doc](state-machine-sender.png)](https://docs.google.com/spreadsheets/d/1o1szGCMx6FzrxfAKV76IEYmeKRJqvV3gd1r3Qfv1Q9o/edit#gid=1176419697)
+
+Note the "optional" reference under the second event in the `await-response` state. That is to indicate that an out of band message might be a single use message with a transition to done, or reusable message (received by many receivers) with a transition back to `await-response`.
+
+The *receiver* state machine is as follows:
+
+[![receiver state machine google doc](state-machine-receiver.png)](https://docs.google.com/spreadsheets/d/1o1szGCMx6FzrxfAKV76IEYmeKRJqvV3gd1r3Qfv1Q9o/edit#gid=10874521)
+
+Worth noting is the first event of the `done` state, where the receiver may receive the message multiple times. This represents, for example, an agent returning to the same website and being greeted with the same QR code each time.
+
+### Messages
+
+There are two messages in the out of band protocol. They are both sent by the *sender*. Only one of the message types is ever used in an instance of the protocol.
+
+#### Message Type: `https://didcomm.org/outofband/VER/invitation`
+
+```jsonc
+{
+  "@type": "https://didcomm.org/outofband/VER/invitation",
+  "@id": "<id used for context as pthid>",
+  "label": "Faber College",
+  "protocols": [
```

Comment:
I'm not sure this is a good idea. See my comment about version negotiation.

Now, having said that, I don't think version negotiation totally solves the problem you're attacking here. It's only about an 80% overlap. Specifically, it doesn't handle the case where a version is renamed. So we may need what you are proposing. I am just noting that I think we need to ponder the similarities in these two mechanisms, and do something to resolve them.

Reply:
I'm guessing you mean the "protocols" field?  If so, I think we're covering the conversation above.